src/platform/packages/shared/kbn-esql-ast/src/commands_registry/location.ts

```diff
@@ -33,6 +33,23 @@ const commandOptionNameToLocation: Record<string, Location> = {
   rerank: Location.RERANK,
 };
 
+/**
+ * Configuration for function-based locations.
+ * Maps command name -> function name -> location config.
+ * If argIndex is specified, the position must be within that specific argument.
+ */
+const functionBasedLocations: Record<
+  string,
+  Record<string, { location: Location; displayName: string; argIndex?: number }>
```

Comment:
Currently the only case is stats where. But I'll keep a map in case we need to add more.